### ISSUE TITLE
Undefined var at level 13 should give error

### ISSUE THREAD

**Issue by Felienne** [Felienne/hedy#1832]
Why doesn't this give an VarUndefined error?

Level 13
```
        name = ask 'what is your name?'
        leeftijd = ask 'what is your age?'
        if name is 'Hedy' and age is 2
            print 'You are the real Hedy!'
```

_Originally posted by @Felienne in https://github.com/Felienne/hedy/pull/1828#r794333432_

**Comment by ThomasKluiters** [comment]
Would like to pick this one up to get a bit more familiar with Hedy!

**Comment by Felienne** [comment]
> Would like to pick this one up to get a bit more familiar with Hedy!

Thanks for offering @ThomasKluiters! Happily assigning this to you, and kudos for picking up a challenging issue as the first one!! 🔥

**Comment by ThomasKluiters** [comment]
I think this is expected behaviour! As we are not required to use quotation marks to indicate strings in level 13 the 'age' is interpreted as a string, so, we are just directly comparing '2' to 'age':

```python
if str(name) == str('Hedy') and str('age') == str('2'):
        print(f'You are the real Hedy!')
```

**Comment by ThomasKluiters** [comment]
@Felienne Should I close this? Or do you think my previous comment is incorrect?

**Comment by Felienne** [comment]
> I think this is expected behaviour! As we are not required to use quotation marks to indicate strings in level 13 the 'age' is interpreted as a string, so, we are just directly comparing '2' to 'age':
> 
> ```python
> if str(name) == str('Hedy') and str('age') == str('2'):
>         print(f'You are the real Hedy!')
> ```
> 

Yeah so I think the real issue is that we should make quotation marks mandatory at this level (or in level 12 makes more sense) See also #1913 

> Though, perhaps an interesting way to "solve" this is simply failing the program if an unused variable exists?

A bit unrelated, I think it is a great idea!! Failing might be a bit harsh, but a warning would be very reasonable!

**Comment by Felienne** [comment]
This too I think might be fixed in #2268? Can you check @boryanagoncharenko?

**Comment by boryanagoncharenko** [comment]
Yes, this is the exact code that had to be changed in many levels. It is addressed now.
